Source/DafnyLanguageServer/Language/CompilerOptions.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.Dafny.LanguageServer.Language {
+  /// <summary>
+  /// Options for the resolution and compilation pipeline
+  /// </summary>
+  public class CompilerOptions {
+    /// <summary>
+    /// The IConfiguration section of the compiler options.
+    /// </summary>
+    public const string Section = "Compiler";
+
+    /// <summary>
+    /// Gets or sets which backends will be targeted so that their resolver can run before verification.
+    /// </summary>
+    public string Backends { get; set; } = "";
```

Comment:
.NET Core's configuration utilities support the usage of arrays. However, they're not as comfortable to use from the command line as your implementation. If I remember right, it'd look something like this:
```sh
--compiler:backends:0=<assembly1> --compiler:backends:1=<assembly2> 
```
How often do you expect this feature to be relevant for end-users? As an alternative, users may also provide (any) language server options through `LanguageServer.appsettings.json`.

Reply:
I don't expect most users to actually interact with the DafnyLanguageServer on the command line.
LanguageServer.appsettings.json. is not used to pass options to the Dafny process itself, so I had to add this option to the language server itself so that it initializes the options of Dafny when resolving a document.

Since by default I don't expect to have more than one assembly, it's better to keep one single parameter. It makes it easy as well for the VSCode extension to offer a single field to provide the argument.